File: rust/mlt-ffi/bindings/java/src/main/java/org/maplibre/mlt/ConvertError.java
package org.maplibre.mlt;

/**
 * Error type returned by FFI conversion functions.
 */
public enum ConvertError {
    /** Input bytes could not be parsed or decoded. */
    INVALID_INPUT(0),
    /** Encoding failed. */
    ENCODING_FAILED(1);

    final int code;

    ConvertError(int code) {
        this.code = code;
    }

    static ConvertError fromCode(int code) {
        return switch (code) {
            case 0 -> INVALID_INPUT;
            case 1 -> ENCODING_FAILED;
            default -> throw new IllegalArgumentException("Unknown ConvertError code: " + code);
        };
    }
}

File: rust/mlt-ffi/bindings/kotlin/src/main/kotlin/org/maplibre/mlt_ffi/ConvertError.kt
package org.maplibre.mlt_ffi

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface ConvertErrorLib: Library {
}
/** Error type returned by FFI conversion functions.
*/
enum class ConvertError {
    InvalidInput,
    EncodingFailed;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<ConvertErrorLib> = ConvertErrorLib::class.java
        internal val lib: ConvertErrorLib = Native.load("mlt_ffi", libClass)
        fun fromNative(native: Int): ConvertError {
            return ConvertError.entries[native]
        }

        fun default(): ConvertError {
            return InvalidInput
        }
    }
}
class ConvertErrorError internal constructor(internal val value: ConvertError): Exception("Rust error result for ConvertError") {
    override fun toString(): String {
        return "ConvertError error with value " + value
    }

    fun getValue(): ConvertError = value
}
